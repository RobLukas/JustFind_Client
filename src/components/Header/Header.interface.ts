interface HeaderProps {
  logo: string;
}

export default HeaderProps;
