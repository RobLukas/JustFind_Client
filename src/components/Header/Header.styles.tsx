import styled from 'styled-components';

const Header = styled.header`
  @media (min-width: 1020px) {
    height: 65px;
  }
  background: ${({ theme }) => theme.styles.secondary.background};
  display: flex;
  align-items: center;
`;

const Logo = styled.img`
  @media (min-width: 1020px) {
    margin: 0 25px 0 15px;
  }
  width: 160px;
  height: 55px;
`;

export default {
  Header,
  Logo,
};
