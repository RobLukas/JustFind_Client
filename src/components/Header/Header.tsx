import React, { FC } from 'react';
import { v4 as uuid } from 'uuid';

import navLinks from '@api/headerNavLinks';
import S from './Header.styles';
import NavLink from './NavLink/NavLink';
import { darkLogo } from '@assets/index';

const Header: FC = () => {
  return (
    <S.Header>
      <S.Logo src={darkLogo} alt={'justfind.pl'} />
      {navLinks.map((navLink) => (
        <NavLink key={uuid()} {...navLink} />
      ))}
    </S.Header>
  );
};

export default Header;
