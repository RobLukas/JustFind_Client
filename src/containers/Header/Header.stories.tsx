import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import { Header } from 'containers';

export default {
  title: 'containers/Header',
  component: Header,
} as Meta;

const Template: Story = () => <Header />;

export const header = Template.bind({});
