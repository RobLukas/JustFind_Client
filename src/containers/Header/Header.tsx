import React, { FC } from 'react';
import { useSelector } from 'react-redux';

import { selectIsDarkMode } from 'features/ThemeMode/themeModeSlice';
import { darkLogo, lightLogo } from 'assets';
import { Header as HeaderComponent } from 'components';

const Header: FC = () => {
  const isDarkMode = useSelector(selectIsDarkMode);
  const logo = isDarkMode ? darkLogo : lightLogo;

  return <HeaderComponent logo={logo} />;
};

export default Header;
