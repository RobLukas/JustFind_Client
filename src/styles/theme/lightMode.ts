export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
