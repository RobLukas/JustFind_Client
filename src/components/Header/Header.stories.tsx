import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import Header from './Header';
import ThemeWrapper from '@containers/ThemeWrapper/ThemeWrapper';

export default {
  title: 'components/Header',
  component: Header,
} as Meta;

const Template: Story = () => (
  <ThemeWrapper>
    <Header />
  </ThemeWrapper>
);

export const header = Template.bind({});
