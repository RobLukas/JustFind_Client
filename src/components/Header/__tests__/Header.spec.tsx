import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import lightMode from 'styles/theme/lightMode';
import Header from '../Header';

describe('<Header />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <Header />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
