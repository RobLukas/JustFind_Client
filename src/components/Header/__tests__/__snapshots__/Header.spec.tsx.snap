// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly 1`] = `
.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 160px;
  height: 55px;
}

.c2 {
  margin: 0 20px 0 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #99a1ab;
}

.c2:hover {
  color: #ff4081;
}

.c2:hover .MuiSvgIcon-root {
  fill: #ff4081;
  -webkit-transition: unset;
  transition: unset;
}

.c3 {
  margin-right: 12px;
}

@media (min-width:1020px) {
  .c0 {
    height: 65px;
  }
}

@media (min-width:1020px) {
  .c1 {
    margin: 0 25px 0 15px;
  }
}

@media (max-width:1020px) {
  .c2 {
    display: none;
  }
}

<header
  class="c0"
>
  <img
    alt="justfind.pl"
    class="c1"
    src="justfind-logo-dark.png"
  />
  <a
    class="c2"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-root c3"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      Job offers
    </span>
  </a>
</header>
`;
